src/Uno.UWP/Devices/Haptics/SimpleHapticsController.Android.cs

```diff
@@ -1,78 +1,83 @@
-				var androidFeedback = FeedbackToAndroidFeedback(feedback);
-#pragma warning disable CA1422 // Validate platform compatibility
-				bool hapticFeedbackEnabled = Settings.System.GetInt(activity.ContentResolver, Settings.System.HapticFeedbackEnabled, 0) != 0;
-#pragma warning restore CA1422 // Validate platform compatibility
-				if (hapticFeedbackEnabled)
-				{
-					var executed = activity.Window?.DecorView.PerformHapticFeedback(androidFeedback) ?? false;
-					if (!executed && PhoneVibrationDevice.GetDefault() is { } vibrationDevice)
-					{
-						// Fall back to VibrationDevice
-						vibrationDevice.Vibrate(feedback.Duration);
-					}
-				}
-			}
-			catch (Exception ex)
-			{
-				if (this.Log().IsEnabled(LogLevel.Error))
-				{
-					this.Log().LogError($"Could not send haptic feedback: {ex}");
-				}
-			}
-		}
+            Vibrator vibrator = (Vibrator)Android.App.Application.Context.GetSystemService(Android.Content.Context.VibratorService);
 
-		private static FeedbackConstants FeedbackToAndroidFeedback(SimpleHapticsControllerFeedback feedback)
-		{
-			if (feedback.Waveform == KnownSimpleHapticsControllerWaveforms.Click)
-			{
-				return FeedbackConstants.ContextClick;
-			}
-			else if (feedback.Waveform == KnownSimpleHapticsControllerWaveforms.Press)
-			{
-				return FeedbackConstants.LongPress;
-			}
-			else
-			{
-				throw new NotSupportedException("Unsupported feedback waveform");
-			}
-		}
-	}
+            if (feedback.Waveform == KnownSimpleHapticsControllerWaveforms.Press)
+            {
+                if (vibrator.HasVibrator)
+                {
+                    vibrator.Vibrate((long)feedback.Duration.TotalMilliseconds);
+                }
+                else
+                {
+                    throw new NotSupportedException("Device does not support vibration");
+                }
+            }
+            else if (feedback.Waveform == KnownSimpleHapticsControllerWaveforms.Click)
+            {
+                if (feedback.Duration.TotalMilliseconds <= 100)
+                {
+                    // Single Click
+                    if (vibrator.HasVibrator)
+                    {
+                        vibrator.Vibrate(VibrationEffect.CreateOneShot((long)feedback.Duration.TotalMilliseconds, VibrationEffect.DefaultAmplitude));
+                    }
+                    else
+                    {
+                        throw new NotSupportedException("Device does not support vibration");
+                    }
+                }
+                else if (feedback.Duration.TotalMilliseconds <= 200)
+                {
+                    // Double Click
+                    if (vibrator.HasVibrator)
+                    {
+                        long[] pattern = { 0, (long)feedback.Duration.TotalMilliseconds, 50, (long)feedback.Duration.TotalMilliseconds };
+                        vibrator.Vibrate(VibrationEffect.CreateWaveform(pattern, -1));
+                    }
```

Comment:
We should be keeping this try/catch, in case there's an unexpected error. Failing haptics should not fail the app.